packages/firestore/src/core/sync_engine.ts

```diff
@@ -915,6 +923,20 @@ export class SyncEngine implements RemoteSyncer, SharedClientStateSyncer {
     return activeQueries;
   }
 
+  private isRetryableTransactionError(error: Error): boolean {
+    if (error.name === 'FirebaseError') {
```

Comment:
Is there a reason not to do an `error instanceof FirestoreError` here? And then you shouldn't need the `as` cast below...

Reply:
Yeah, I initially used `error instanceof FirestoreError`, but the error that is surfaced for the transaction tests `increment transactionally` and `update transactionally` fail that check:
```
 ✗ increment transactionally
	FirebaseError: the stored version (1564089858307124) does not match the required base version (1564089858018004)
	    at new FirestoreError (test/integration/bootstrap.js:16790:11599)
	    at X.<anonymous> (test/integration/bootstrap.js:16331:45296)
	    at nb (test/integration/bootstrap.js:14538:195)
	    at X.../webchannel-wrapper/dist/index.esm.js.g.dispatchEvent (test/integration/bootstrap.js:14536:304)
	    at vd (test/integration/bootstrap.js:14563:390)
	    at xd (test/integration/bootstrap.js:14566:62)
	    at X.../webchannel-wrapper/dist/index.esm.js.g.$a (test/integration/bootstrap.js:14564:363)
	    at X.../webchannel-wrapper/dist/index.esm.js.g.Fa (test/integration/bootstrap.js:14564:339)
```

I'm not sure why this is the case, since I am throwing a new Firestore error (from the most recent webchannel PR).

Reply:
Huh, interesting...  I'm going to guess it's related to this: https://stackoverflow.com/a/30403029

(the way transpiled inheritance works and the way the native Error object is implemented are incompatible or something)